src/v/raft/consensus.cc

```diff
@@ -1335,6 +1328,13 @@ ss::future<vote_reply> consensus::do_vote(vote_request&& r) {
         return ss::make_ready_future<vote_reply>(std::move(reply));
     }
 
+    /// set to true if the caller's log is as up to date as the recipient's
+    /// - extension on raft. see Diego's phd dissertation, section 9.6
+    /// - "Preventing disruptions when a server rejoins the cluster"
+    reply.log_ok
+      = r.prev_log_term > last_entry_term
+        || (r.prev_log_term == last_entry_term && r.prev_log_index >= last_log_index);
+
```

Comment:
ahhh, I think this change partially addresses the question on the previous commit.